dashboard/src/shared/AxiosInstance.ts

```diff
@@ -25,15 +25,25 @@ export function createAxiosInterceptors(axios: AxiosInstance, store: Store<IStor
   axios.interceptors.response.use(
     response => response,
     e => {
+      const dispatch = store.dispatch as ThunkDispatch<IStoreState, null, Action>;
       const err: AxiosError = e;
+      if (
+        err.config.xsrfCookieName === "XSRF-TOKEN" &&
+        err.code === undefined &&
+        err.message === "Network Error" &&
+        !err.response &&
+        Auth.usingOIDCToken()
+      ) {
+        // The OIDC token is no longer valid, logout
+        dispatch(actions.auth.logout());
```

Comment:
why don't we dispatch a new action called `expireSession` or something? That way we can have a specific field in the state (`sessionExpired`) that deals with this case? Then, PrivateRoute is simpler and just needs to render the modal if `sessionExpired` is true.

Reply:
What I was thinking was having two actions, to separate the concerns of the two cases:

- actions.auth.logout - untouched, does what it does today to clear the token and set authenticated to false
- actions.auth.expireSession - sets sessionExpired to true to bring up the modal and dispatches actions.auth.logout

If we separate them in this way, we shouldn't have the issue where we show the modal if a user clicks logout, right?

Reply:
Yes, as far as we call `expireSession` when we receive a 401

Reply:
@andresmgot we used to show an error message on the login page when we received a 401 right? Do we still need that if we have a modal now?

Reply:
Yes, that message (and the form) is showed when not using an OIDC token

Reply:
@andresmgot ah yes, you're right - thanks